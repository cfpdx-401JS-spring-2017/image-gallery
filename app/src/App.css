.App {
  text-align: center;
  display: flex;
  height: 100vh;
  width: 100%;
  justify-content: center;
  align-items: center;

}

.ThumbNail {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  height: 30vh;
}

.List {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

button {
  height: 10vh;
  width: 20vw;
  background-color: #476A6F;
  border: none;
  margin: 2vh 2vh;
  box-shadow: 1px 1px 1px 1px #202A25;
  border-radius: 10px;
}

.Link {
  color: #F7F0F0;
  text-decoration: none;
  font-size: 2em;
}

.Link:hover {
  
}
