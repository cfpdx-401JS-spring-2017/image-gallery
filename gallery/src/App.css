.App {
  text-align: center;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
}

.ThumbNail {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  height: 30vh;
}

.List {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

button {
  height: 8vh;
  width: 12vw;
  background-color: #2F3E46;
  color:#CAD2C5;
  border: none;
  margin: 2vh 2vh;
  box-shadow: none;
}
