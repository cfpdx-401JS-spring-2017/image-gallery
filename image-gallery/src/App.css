.App {
  text-align: center;
}


.App-intro {
  font-size: large;
}

.header-container {
  text-align: center;
  font-family: 'Plaster', cursive;
  /*font-family: 'Montserrat Subrayada', sans-serif;*/
  font-size: 20px;
}

.header-title:hover {
  cursor: pointer;
  color: rgb(255, 82, 82);

}

.brighten img {
  max-width: 35%;
  border-radius: 15px;
-webkit-filter: brightness(110%);
-webkit-transition: all 1s ease;
-moz-transition: all 1s ease;
-o-transition: all 1s ease;
-ms-transition: all 1s ease;
transition: all 1s ease;
position: absolute;


}

.brighten img:hover{
-webkit-filter: brightness(175%);

}

span.play-video {
    display: none;
    position: relative;
    text-align: center;
   
}

span.play-video:hover, a.video:hover + span {
    display: block;
}


.view-buttons {
  margin: 15px;
  padding: 10px;
  font-size: 15px;
  border-radius: 5px;
  color: grey;
  font-weight: bolder;
  border: 1px solid rgb(255, 82, 82);
  transition-duration: 0.4s;
  font-family: 'Raleway', sans-serif;
  background-color: white;

}

.view-buttons:hover {
  background-color: rgb(255, 82, 82);
  color: white;
  cursor: pointer;
}

.images{
   max-width: 40%;
   font-family: Courier New, Courier, monospace;
}

.list-view{
  list-style: none;

}

/*.bottom-list-item {
  font-family: 'Raleway', sans-serif;
  font-size: 15px;
  border-bottom: 2px, solid, black;
}*/

.image-caption{
 font-family: 'Raleway', sans-serif;
 font-size: 15px;
}


.arrow:hover {
 color: rgb(255, 82, 82);
}

.thumbnail-image:hover {
  cursor: pointer;
  transform: scale(1.5); 

}



/* Positioning help: https://jsfiddle.net/6k2Lhaje/7/ */
/* Transition brightening help: http://www.thebrandsmen.com/css-image-hover-effects/ */



