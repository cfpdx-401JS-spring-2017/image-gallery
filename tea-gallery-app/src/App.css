@import url('https://fonts.googleapis.com/css?family=Josefin+Sans');

body {
  margin: 0 auto;
  padding: 0;
  font-family: 'Josefin Sans', sans-serif;
}

li {
  list-style-type: none;
}

button {
  height: 30px;
  width: 70px;
  margin: 0;
  border: 1px solid grey;
  border-radius: 15px;
  background-color: hsla(176, 13%, 68%, 0.56);
}

button:hover {
  background-color: hsla(176, 61%, 68%, 0.56);
  cursor: pointer;
}

/*nav*/
.nav {
  text-align: center;
}

.nav button {
  width: 12%;
  height: 70px;
  margin: 1%;
  font-size: .75em;
}

.wrapper {
  text-align: center;
}

.app-header {
  text-align: center;
  font-size: 1.3em;
}

.app-header h2, h3 {
  margin: 2%;
}

/*albums styles*/
.albums-view {
  text-align: center;
}

.albums-view button {
  width: 20%;
  height: 60px;
  margin: 2% 5% 0 5%;
  font-size: 1em;
  padding: 5px;
}

.albums-view button h3 {
  margin: 0;
}

.view-button-wrapper {
  margin: 2%;
}

.view-button-wrapper button {
  margin: 1%;
}

/*Thumbnail styles*/

.thumbnail-list { /*why is there a mysterious padding-left?*/
  max-width: 90%;
  margin: 0 auto;
}

.thumbnail-wrapper {
  display: inline-block;
  width: 30%;
}

.thumbnail-li {
  padding: 0 auto;
}

.thumbnail {
  width: 90%;
  margin: 8% auto;
}

/*List styles*/

.list-item-wrapper {
  border: 1px solid grey;
  border-radius: 10px;
  padding: 1%;
  width: 25%;
  margin: 2%;
  text-align: left;
    display: inline-block;
}

.list-item-wrapper button {
  margin: 0 auto; 
  /*doesmt work*/
}

.image-list-item {
  margin: .75%;
}

/*Gallery styles*/

.gallery-img-wrapper {
  width: 60%;
  margin: 0 auto;
}

.gallery-img {
  width: 100%;
}

.next-button {
  float: right;
}

.back-button {
  float: left;
}

/*Add and delete styling*/

.add-item {
  margin: 5% auto;
  font-size: 2em;
  width: 60%;
  font-size: 20px;
}

.add-item input {
  height: 30px;
  margin: 0 10px 10px 10px;
  font-size: 20px;
}

.add-item button {
  width: 100px;
  height: 40px;
  border: 1px solid black;
  border-radius: 10px;
}
