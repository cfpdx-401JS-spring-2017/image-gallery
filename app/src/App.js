import React, { Component } from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect
} from 'react-router-dom';

import Home from './components/Home';
import Albums from './components/albums/Albums';
import './styles/App.css';

class App extends Component {
  render() {
    return (
      <Router>
        <div className="App">
          <h2>CPDB: Cute Pomeranian Database</h2>
          <p>
            Welcome to the <code>Cute Pomeranian Database</code>! Check out our poms by clicking below:
          </p>
          <Switch>
            <Route exact path='/' component={Home}/>
            <Route path='/albums' component={Albums} />
            <Redirect to='/' />
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;
