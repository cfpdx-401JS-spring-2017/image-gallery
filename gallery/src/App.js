import React, { Component } from 'react';
import './App.css';
import AlbumsViewer from './AlbumsViewer';
import Albums from './Albums';

import {
  BrowserRouter as Router,
  Route, Switch, Redirect, Link
} from 'react-router-dom'

class App extends Component {
  constructor() {
    super();
    this.state = {
    }

  }

  render() {
  
    return (
      <div className="App">
        <Router>
            <div>
                <Switch>
                    {/*<Route exact path="/" component={App}/>*/}
                    <Route path="/albums/:id" component={AlbumsViewer}/>
                    <Route path="/albums" component={Albums}/>
                    <Redirect path="/albums" component={Albums}/>
                </Switch>
                <button>
                  <Link to="/albums">Go to Albums</Link>
                </button>
            </div>
        </Router>
      </div>  
    );
  }
}

export default App;
