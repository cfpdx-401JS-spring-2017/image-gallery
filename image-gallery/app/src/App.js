import React from 'react';
import './App.css';
import Header from './components/Header';
import Intro from './components/IntroView';
import Albums from './components/Albums'
import { BrowserRouter as Router, Route, Switch, Redirect } from 'react-router-dom';



export default function App() {   
    return (
      <Router>
        <div>
          <Header />

          <Switch>
            <Route exact path='/' component={Intro} />
            <Route path='/albums' component={Albums} />
            <Redirect to='/'/>
            
          </Switch>
        </div>
      </Router>

    );
  }


