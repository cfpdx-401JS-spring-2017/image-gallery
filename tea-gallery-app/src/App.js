import React from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Link, 
  Redirect
} from 'react-router-dom';
import './App.css';
import AlbumsContainer from './components/AlbumsContainer';

export default function App(props) {

  return (
    <Router>
      <div className="App" >
        <div className="app-header">
          <h2>Welcome to the Cha'llery</h2>
          <h3>A photo gallery of <strong>èŒ¶</strong> --> 'Tea'</h3>
        </div>
        <div className="nav">
          <Link to="/albums">
            <button>
              <h3>Albums</h3>
            </button>
          </Link>
          <Link to="/">
            <button>
              <h3>Home</h3>
            </button>
          </Link>
        </div>
        <Switch>
          <Route path="/albums" component={AlbumsContainer} />
          <Redirect to="/"/>;
        </Switch>
      </div >
    </Router>
  );
}


