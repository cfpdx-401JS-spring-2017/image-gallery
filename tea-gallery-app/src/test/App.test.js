import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import ReactDOM from 'react-dom';

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
  });

  it('renders the app', () => {
    const component = renderer.create(<App />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot(); 
  });

});
