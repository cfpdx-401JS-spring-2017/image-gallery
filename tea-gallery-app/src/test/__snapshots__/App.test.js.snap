// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the app 1`] = `
<div
  className="App"
>
  <div
    className="app-header"
  >
    <h2>
      Welcome to the Cha'llery
    </h2>
    <h3>
      A photo gallery of 
      <strong>
        èŒ¶
      </strong>
       --&gt; 'Tea'
    </h3>
  </div>
  <div
    className="nav"
  >
    <a
      href="/albums"
      onClick={[Function]}
    >
      <button>
        <h3>
          Albums
        </h3>
      </button>
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <button>
        <h3>
          Home
        </h3>
      </button>
    </a>
  </div>
</div>
`;
